lib/Map/Vector/Protomaps/ProtomapsGeojsonSource.ts

```diff
@@ -0,0 +1,285 @@
+      if (!feature.bbox) {
+        feature.bbox = bbox(feature);
+      }
+
+      // Filter by bounding box and then intersection with buffer (to minimize calls to booleanIntersects)
+      if (
+        Math.max(
+          feature.bbox[0],
+          // Wrap buffer bbox if necessary
+          feature.bbox[0] > 180 ? bufferBbox[0] + 360 : bufferBbox[0]
+        ) <=
+          Math.min(
+            feature.bbox[2],
+            // Wrap buffer bbox if necessary
+            feature.bbox[2] > 180 ? bufferBbox[2] + 360 : bufferBbox[2]
+          ) &&
+        Math.max(feature.bbox[1], bufferBbox[1]) <=
+          Math.min(feature.bbox[3], bufferBbox[3])
+      ) {
+        // If we have longitudes greater than 180 - used wrappedBuffer
+        if (feature.bbox[0] > 180 || feature.bbox[2] > 180) {
+          if (booleanIntersects(feature, wrappedBuffer))
+            pickedFeatures.push(feature);
+        } else if (booleanIntersects(feature, buffer))
+          pickedFeatures.push(feature);
+      }
+    }
+
+    // Convert pickedFeatures to ImageryLayerFeatureInfos
+    pickedFeatures.forEach((f) => {
+      const featureInfo = new ImageryLayerFeatureInfo();
+
+      featureInfo.data = f;
+      featureInfo.properties = Object.assign(
+        { [LAYER_NAME_PROP]: GEOJSON_SOURCE_LAYER_NAME },
+        f.properties ?? {}
+      );
+
+      if (
+        f.geometry.type === "Point" &&
+        typeof f.geometry.coordinates[0] === "number" &&
+        typeof f.geometry.coordinates[1] === "number"
+      ) {
+        featureInfo.position = Cartographic.fromDegrees(
+          f.geometry.coordinates[0],
+          f.geometry.coordinates[1]
+        );
+      }
+
+      featureInfo.configureDescriptionFromProperties(f.properties);
+      featureInfo.configureNameFromProperties(f.properties);
+
+      featureInfos.push(featureInfo);
+    });
+
+    return featureInfos;
+  }
+}
+
+export const geomTypeMap = (
+  type: string | null | undefined
+): GeomType | null => {
+  switch (type) {
+    case "Point":
+      return GeomType.Point;
+    case "LineString":
+      return GeomType.Line;
+    case "Polygon":
+      return GeomType.Polygon;
+    case "MultiPoint":
+      return GeomType.Point;
+    case "MultiLineString":
+      return GeomType.Line;
+    case "MultiPolygon":
+      return GeomType.Line;
+    default:
+      return null;
+  }
+};
+
+export function geojsonVtTileToProtomapsFeatures(
+  features: geojsonvt.Feature[],
+  tileSize: number
+): ProtomapsFeature[] {
+  const scale = tileSize / GEOJSON_VT_EXTENT;
+
+  return features
+    .map((f) => {
+      let transformedGeom: Point[][] = [];
+      let numVertices = 0;
+
+      // Calculate bbox
+      const bbox: Bbox = {
+        minX: Infinity,
+        minY: Infinity,
+        maxX: -Infinity,
+        maxY: -Infinity
+      };
+
+      const geom = f.geometry;
+      transformedGeom = [
+        geom.map((g1) => {
+          g1 = [g1[0] * scale, g1[1] * scale];
```

Comment:
Is the LHS `g1` conceptually a pair `(x, y)`?  I think it would be easier to read separate  `x` and `y` variables instead of `g1[0]` and `g1[1]`